pyfar/dsp/dsp.py

```diff
@@ -227,6 +227,202 @@ def spectrogram(signal, dB=True, log_prefix=20, log_reference=1,
+    the indexes of the samples given in `length` denote the first sample of
+    the windowing which is non-zero and the first being one. For the right
+    side of a symmetric window and for ``shape='right'``, the samples denote
+    the last sample being one and the last being non-zero.
+
+    This function calls `scipy.signal.windows.get_window` to create the
+    window.
+    Window types:
+    - ``boxcar``
+    - ``triang``
+    - ``blackman``
+    - ``hamming``
+    - ``hann``
+    - ``bartlett``
+    - ``flattop``
+    - ``parzen``
+    - ``bohman``
+    - ``blackmanharris``
+    - ``nuttall``
+    - ``barthann``
+    - ``kaiser`` (needs beta, see :py:func:`~pyfar.dsp.kaiser_window_beta`)
+    - ``gaussian`` (needs standard deviation)
+    - ``general_gaussian`` (needs power, width)
+    - ``dpss`` (needs normalized half-bandwidth)
+    - ``chebwin`` (needs attenuation)
+    - ``exponential`` (needs center, decay scale)
+    - ``tukey`` (needs taper fraction)
+    - ``taylor`` (needs number of constant sidelobes,
+      sidelobe level)
+    If the window requires no parameters, then `window` can be a string.
+    If the window requires parameters, then `window` must be a tuple
+    with the first argument the string name of the window, and the next
+    arguments the needed parameters.
+    """
+    # Check input
+    if not isinstance(signal, pyfar.Signal):
+        raise TypeError("The parameter signal has to be of type: Signal.")
+    if shape not in ('symmetric', 'left', 'right'):
+        raise ValueError(
+            "The parameter shape has to be 'symmetric', 'left' or 'right'.")
+    if not isinstance(truncate, bool):
+        raise TypeError("The parameter truncate has to be of type: bool.")
+    if not isinstance(length, (list, type(None))):
+        raise TypeError(
+            "The parameter length has to be of type list or None.")
+
+    # Check length
+    if length is not None:
+        if length != sorted(length):
+            raise ValueError("Values in length need to be in ascending order.")
+        # Convert to samples
+        if unit == 's':
+            length = [round(li*signal.sampling_rate) for li in length]
+        elif unit == 'ms':
+            length = [round(li*signal.sampling_rate/1e3) for li in length]
+        elif unit == 'samples':
+            length = [int(li) for li in length]
+        else:
+            raise ValueError(f"unit is {unit} but has to be"
+                             f" 'samples', 's' or 'ms'.")
+        # Check window size
+        if length[-1] > signal.n_samples:
+            raise ValueError(
+                "Values in length require window to be longer than signal.")
+
+    # Create window
+    # win_start and win_stop define the first and last sample of the window
+    if length is None:
+        win_samples = signal.n_samples
+        win = sgn.windows.get_window(window, win_samples, fftbins=False)
+        win_start = 0
+        win_stop = signal.n_samples-1
```

Comment:
I don't see so many use-cases for this one, how about you others?

Reply:
Agree

Reply:
I see at least some use cases, and the current syntax would not allow to create a **pure** symmetric window (because the second and first value for ``length``/``interval`` denote the sample with a value of ``1``.  How about this:

Reply:
The problem with an interval of one value and a center parameter is that that the center would need o be between two samples for an even lengh of a window (a pure symmetric). Therefore I suggest to remove interval=None, and instead use the shape='symmetric' opion with start and end, which automatically creates a window with even(pure symmetric) or odd (value at center is 1) window depending on the input.